scapy/sendrecv.py

```diff
@@ -853,7 +853,11 @@ def prn(pkt):
 
 @conf.commands.register
 def tshark(*args,**kargs):
-    """Sniff packets and print them calling pkt.show(), a bit like text wireshark"""
-    sniff(prn=lambda x: x.display(),*args,**kargs)
-
-
+    """Sniff packets and print them calling pkt.summary(), a bit like text wireshark"""
+    print("Capturing on '" + str(kargs.get('iface') if 'iface' in kargs else conf.iface) + "'")
+    i = [0]  # Must be editable from inside a callback
```

Comment:
Did you try using `global` with a regular integer ?

Reply:
Yes, but it didn't work. Doing such a simple thing is always a mess...

Reply:
Here would have been a good place to use `nonlocal`, but that's Python 3 only :-/. Instead of this comment, could you write something like "This should be a nonlocal variable, using a mutable object for Python 2 compatibility"? This way, we will spot it when Python 2.7 eventually disappears.